docs/csharp/language-reference/compiler-options/advanced.md

```diff
@@ -0,0 +1,301 @@
+
+1. Current working directory.
+1. The common language runtime system directory.
+1. Directories specified by **AdditionalLibPaths**.
+1. Directories specified by the LIB environment variable.
+
+Use **Reference** to specify an assembly reference. **AdditionalLibPaths** is additive. Specifying it more than once appends to any prior values. Since the path to the dependent assembly isn't specified in the assembly manifest, the application will find and use the assembly in the global assembly cache. The compiler referencing the assembly doesn't imply the common language runtime can find and load the assembly at runtime. See [How the Runtime Locates Assemblies](../../../framework/deployment/how-the-runtime-locates-assemblies.md) for details on how the runtime searches for referenced assemblies.  
+
+## GenerateFullPaths
+
+The **GenerateFullPaths** option causes the compiler to specify the full path to the file when listing compilation errors and warnings.
+  
+```Xml
+<GenerateFullPaths>true</GenerateFullPaths>
+```
+
+By default, errors and warnings that result from compilation specify the name of the file in which an error was found. The **GenerateFullPaths** option causes the compiler to specify the full path to the file. This compiler option is unavailable in Visual Studio and cannot be changed programmatically.
+
+## PreferredUILang
+
+By using the **PreferredUILang** compiler option, you can specify the language in which the C# compiler displays output, such as error messages.
+
+```xml
+<PreferredUILang>language</PreferredUILang>
+```
+
+Where `language` is the [language name](/windows/desktop/Intl/language-names) of the language to use for compiler output. You can use the **PreferredUILang** compiler option to specify the language that you want the C# compiler to use for error messages and other command-line output. If the language pack for the language isn't installed, the language setting of the operating system is used instead.
+
+## BaseAddress
+
+The **BaseAddress** option lets you specify the preferred base address at which to load a DLL. For more information about when and why to use this option, see [Larry Osterman's WebLog](/archive/blogs/larryosterman/why-should-i-even-bother-to-use-dlls-in-my-system).
+
+```xml
+<BaseAddress>address</BaseAddress>
+```
+
+Where `address` is the base address for the DLL. This address can be specified as a decimal, hexadecimal, or octal number. The default base address for a DLL is set by the .NET common language runtime. The lower-order word in this address will be rounded. For example, if you specify `0x11110001`, it will be rounded to `0x11110000`. To complete the signing process for a DLL, use SN.EXE with the -R option.
+
+## ChecksumAlgorithm
```

Comment:
This could go to the security group, but it may not be worth the effort to move it.

Reply:
I'll keep it here to be consistent with the roslyn docs.